### PR TITLE
Actual ZoneNum Followup

### LINKED ISSUE #9570: Follow up to #9553
Issue overview
--------------
Several clean up tasks are identified while removing ActualZoneNum.

 - Move all baseboards ZonePtr initialization in init to getInput
 - condense multiple Tstat loops in Furnace (and HVACMultiSpeedHeatPump could use same code if it were a function, and HVACVariableRefrigerantFlow?)
 - several class references were noted. Could reduce a lot more code with reference to array. ElectricBaseboard init
 - search for state.dataZoneEquip->ZoneEquipConfig and see if there are other code reducing techniques to compact code
 - could change more ControlledZoneNum to just ZoneNum

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by mjwitte**
Pull request overview
---------------------
 - Fixes #9570 
 - Various code cleanups that were identified in review of #9553 
 - And some more cleanup beyond that

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [x] Perform a Code Review on GitHub
 - [x] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [x] CI status: all green or justified
 - [x] Check that performance is not impacted (CI Linux results include performance check)
 - [x] Run Unit Test(s) locally
 - [x] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/AirflowNetwork/src/Solver.cpp
Comment (mjwitte) [inline src/EnergyPlus/AirflowNetwork/src/Solver.cpp:10345]: Moved some random `Array1D`s to `EPVector`.

File: src/EnergyPlus/BaseboardElectric.cc
Comment (mjwitte) [inline src/EnergyPlus/BaseboardElectric.cc:332]: Use existing function.

File: src/EnergyPlus/BaseboardElectric.cc
Comment (mjwitte) [inline src/EnergyPlus/BaseboardElectric.cc:415]: Moved error message into `GetZoneEquipControlledZoneNum` function which is used during input processing, so this init check is no longer needed.

File: src/EnergyPlus/BaseboardRadiator.cc
Comment (mjwitte) [inline src/EnergyPlus/BaseboardRadiator.cc:266]: Did some extra cleanup for this baseboard type.

File: src/EnergyPlus/BaseboardRadiator.hh
Comment (mjwitte) [inline src/EnergyPlus/BaseboardRadiator.hh:109]: Move these to member methods but just in this baseboard type.

File: src/EnergyPlus/BranchNodeConnections.hh
Comment (mjwitte) [inline src/EnergyPlus/BranchNodeConnections.hh:72]: Moved these name arrays here from DataLoopNode (but didn't move the enum yet because that'll touch every file that manages HVAC nodes.

File: src/EnergyPlus/DataBranchNodeConnections.hh
Comment (mjwitte) [inline src/EnergyPlus/DataBranchNodeConnections.hh:72]: Moved `CompSets` object types from string to enum. I see some commented lines here that need to be deleted.

File: src/EnergyPlus/DataLoopNode.hh
Comment (mjwitte) [inline src/EnergyPlus/DataLoopNode.hh:152]: `ConnectionObjectType` enum was missing some components that do CompSets but don't manage any nodes directly.

File: src/EnergyPlus/DataZoneEquipment.cc
Comment (mjwitte) [inline src/EnergyPlus/DataZoneEquipment.cc:1315]: Error message moved here instead of scattered about in baseboard init functions. Other HVAC components can probably use similar changes.

File: src/EnergyPlus/IntegratedHeatPump.cc
Comment (mjwitte) [inline src/EnergyPlus/IntegratedHeatPump.cc:1233]: Need enums for calls to compsets. Should address earlier comment about naming for enums.

File: src/EnergyPlus/SimulationManager.cc
Comment (mjwitte) [inline src/EnergyPlus/SimulationManager.cc:2616]: Prune dead coe.

File: src/EnergyPlus/SystemReports.cc
Comment (mjwitte) [inline src/EnergyPlus/SystemReports.cc:2193]: The move to component type enums cascaded into many changes in SystemReports.

File: src/EnergyPlus/UnitarySystem.cc
Comment (mjwitte) [inline src/EnergyPlus/UnitarySystem.cc]: Since this wasn't being used, replaced it with two new enums (one for zone equip and one for airloop). @rraustad

File: src/EnergyPlus/UnitarySystem.cc
Comment (mjwitte) [inline src/EnergyPlus/UnitarySystem.cc]: This section duplicated search already done above in `searchTotalComponents`.

File: src/EnergyPlus/UnitarySystem.cc
Comment (mjwitte) [inline src/EnergyPlus/UnitarySystem.cc]: New comptype for airloop unitary systems.

File: src/EnergyPlus/UnitarySystem.cc
Comment (mjwitte) [inline src/EnergyPlus/UnitarySystem.cc]: New comptype for zone equipment unitary systems. Would be nice if these worked off a common enum list, but they don't yet, maybe never, not sure what's the best plan.

File: src/EnergyPlus/UnitarySystem.cc
Comment (mjwitte) [inline src/EnergyPlus/UnitarySystem.cc]: Check enum first, then check name.

**Review by mjwitte — COMMENTED** [review]
This has gone far enough for now (maybe too far). Not planning to touch the furnace code.

Ready for initial review.

File: src/EnergyPlus/UnitarySystem.cc
Comment (rraustad) [reply src/EnergyPlus/UnitarySystem.cc]: These changes don't appear to be a conflict issue with #9627, we'll see.

**Comment by rraustad** [comment]
@mjwitte want me to take a stab at conflicts?

**Comment by mjwitte** [comment]
> @mjwitte want me to take a stab at conflicts?

@rraustad Sure, thanks!

**Comment by rraustad** [comment]
@Myoldmopar this is the final code changes for this branch. The remaining unanswered comment I am aware of is to group the air loop data here similar to what was done with PlantLocation.

    bool searchTotalComponents(EnergyPlusData &state,
                               SimAirServingZones::CompType compTypeToFind,
                               std::string_view objectNameToFind,
                               int &compIndex,
                               int &branchIndex,
                               int &airLoopIndex)

**Comment by amirroth** [comment]
> @Myoldmopar this is the final code changes for this branch. The remaining unanswered comment I am aware of is to group the air loop data here similar to what was done with PlantLocation.
> 
> ```
> bool searchTotalComponents(EnergyPlusData &state,
>                            SimAirServingZones::CompType compTypeToFind,
>                            std::string_view objectNameToFind,
>                            int &compIndex,
>                            int &branchIndex,
>                            int &airLoopIndex)
> ```

That comment had a 📌 in it. Any comment that has that is basically "for later", something I noticed and wanted to call attention to (so I didn't forget) but not something that needs to be dealt with in the current PR.

File: src/EnergyPlus/InputProcessing/InputProcessor.hh
Comment (rraustad) [inline src/EnergyPlus/InputProcessing/InputProcessor.hh:198]: I don't recall the rule on string_view arguments but I believe this is correct.

File: src/EnergyPlus/NodeInputManager.cc
Comment (rraustad) [inline src/EnergyPlus/NodeInputManager.cc]: Doesn't the connection type default to invalid? And isn't thisConnectionType always equal to nodeConnectionType (except for Num which should never happen)? So why is this switch even needed? and the extra local var?

**Comment by rraustad** [comment]
This one is at the finish line.

File: src/EnergyPlus/InputProcessing/InputProcessor.hh
Comment (amirroth) [reply src/EnergyPlus/InputProcessing/InputProcessor.hh:198]: 👍

File: src/EnergyPlus/NodeInputManager.cc
Comment (mjwitte) [reply src/EnergyPlus/NodeInputManager.cc]: This must be leftover from the conversion from string to enum. It's gone now. And the optional_bool `IncrementFluidStream` is now a simple bool. Need to wait for CI to ensure nothing got broken here.

File: src/EnergyPlus/NodeInputManager.cc
Comment (rraustad) [inline src/EnergyPlus/NodeInputManager.cc]: thisConnectionType should show up as an unused variable warning now.

**Review by rraustad — APPROVED** [review]
Reviewed, then reviewed again, and then reviewed. This looks good to go.

**Comment by rraustad** [comment]
Waiting on CI before merging.

**Comment by rraustad** [comment]
SolarShadingTestGPU is the only diff I see so far. I'll wait for CI to complete before merging.

**Pull request merged**
